Sources/MessagingPush/PushHandling/NotificationCenterDelegate.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import UserNotifications
+
+protocol PushEventListener {
+    var delegate: UNUserNotificationCenterDelegate { get }
+
+    func onPushClicked(_ push: PushNotification, completionHandler: @escaping () -> Void)
+    func shouldDisplayPushAppInForeground(_ push: PushNotification, completionHandler: @escaping (UNNotificationPresentationOptions) -> Void)
+
+    func newNotificationCenterDelegateSet(_ newDelegate: UNUserNotificationCenterDelegate?)
+    func beginListening()
+}
+
+struct PushNotification {
+    let pushId: String
+    let deliveryDate: Date
+    let title: String
+    let message: String
+    let data: [AnyHashable: Any]
+
+    init(notification: UNNotification) {
```

Comment:
Parse classes inside of `UserNotifications` framework into classes that we can create instances of.